source: https://hg.mozilla.org/releases/mozilla-esr45/rev/bf0dd9ae6807
gentoo bugzilla: https://bugs.gentoo.org/show_bug.cgi?id=603622
patch fixes trying to enter google and youtube

--

 Bug 1290037 - Update keybits in H2. r=mt, a=ritu
author	Franziskus Kiefer <franziskuskiefer@gmail.com>
	Thu, 28 Jul 2016 16:48:00 +0200 (10 months ago)
changeset 312567 	bf0dd9ae6807
parent 312566 	ecf919481e99
child 312568 	576f03e362c5
push id	373
push user	ryanvm@gmail.com
push date	2017-01-12 03:45 +0000
treeherder	mozilla-esr45@4d82e7314a72 [default view] [failures only]
perfherder	[talos] [build metrics] [platform microbench] (compared to previous push)
reviewers	mt, ritu
bugs	1290037
milestone	45.6.1
Bug 1290037 - Update keybits in H2. r=mt, a=ritu


diff --git a/netwerk/protocol/http/Http2Session.cpp b/netwerk/protocol/http/Http2Session.cpp
--- a/netwerk/protocol/http/Http2Session.cpp
+++ b/netwerk/protocol/http/Http2Session.cpp
@@ -3516,18 +3516,18 @@ Http2Session::ConfirmTLSProfile()
     RETURN_SESSION_ERROR(this, INADEQUATE_SECURITY);
   }
 
   uint32_t keybits = ssl->GetKEAKeyBits();
   if (kea == ssl_kea_dh && keybits < 2048) {
     LOG3(("Http2Session::ConfirmTLSProfile %p FAILED due to DH %d < 2048\n",
           this, keybits));
     RETURN_SESSION_ERROR(this, INADEQUATE_SECURITY);
-  } else if (kea == ssl_kea_ecdh && keybits < 256) { // 256 bits is "security level" of 128
-    LOG3(("Http2Session::ConfirmTLSProfile %p FAILED due to ECDH %d < 256\n",
+  } else if (kea == ssl_kea_ecdh && keybits < 224) { // see rfc7540 9.2.1.
+    LOG3(("Http2Session::ConfirmTLSProfile %p FAILED due to ECDH %d < 224\n",
           this, keybits));
     RETURN_SESSION_ERROR(this, INADEQUATE_SECURITY);
   }
 
   int16_t macAlgorithm = ssl->GetMACAlgorithmUsed();
   LOG3(("Http2Session::ConfirmTLSProfile %p MAC Algortihm (aead==6) %d\n",
         this, macAlgorithm));
   if (macAlgorithm != nsISSLSocketControl::SSL_MAC_AEAD) {





